### PR TITLE
[WIP] reconcile vspheremachine status when vspherevm already exists

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/hold

**Release note**:

```release-note

```

### CODE REVIEW

File: controllers/vspheremachine_controller.go [hunk 1]
```diff
@@ -340,7 +354,7 @@ func (r machineReconciler) reconcileNormal(ctx *context.MachineContext) (reconci
 				"unexpected error while reconciling provider ID for %s", ctx)
 		}
 		ctx.Logger.Info("provider ID is not reconciled")
-		return reconcile.Result{}, nil
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```
Comment (akutz) [inline controllers/vspheremachine_controller.go]: This is a hack and should be avoided long-term. Please add a `// TODO` to indicate this should be revisited.

File: controllers/vspheremachine_controller.go [hunk 2]
```diff
@@ -350,7 +364,7 @@ func (r machineReconciler) reconcileNormal(ctx *context.MachineContext) (reconci
 				"unexpected error while reconciling network for %s", ctx)
 		}
 		ctx.Logger.Info("network is not reconciled")
-		return reconcile.Result{}, nil
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```
Comment (akutz) [inline controllers/vspheremachine_controller.go]: This is a hack and should be avoided long-term. Please add a `// TODO` to indicate this should be revisited.

File: controllers/vspheremachine_controller.go [hunk 3]
```diff
@@ -360,7 +374,7 @@ func (r machineReconciler) reconcileNormal(ctx *context.MachineContext) (reconci
 				"unexpected error while reconciling ready state for %s", ctx)
 		}
 		ctx.Logger.Info("ready state is not reconciled")
-		return reconcile.Result{}, nil
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```
Comment (akutz) [inline controllers/vspheremachine_controller.go]: This is a hack and should be avoided long-term. Please add a `// TODO` to indicate this should be revisited.

File: controllers/vspheremachine_controller.go (hunk 1 above)
Comment (yastij) [reply controllers/vspheremachine_controller.go]: If we're not able to read the providerID from bios-uuid we'd have to wait for the resync period of 10min to reconcile this.

**Pull request closed without merging**